hw/ip/spi_device/data/spi_device_testplan.hjson

```diff
@@ -146,5 +165,78 @@
+            - If hw_reg_dis == 0, the data is returned to the host via return-by-HW register, else the data is returned via read FIFO.
+            - Confirm that the TPM submodule sends WAIT until the read FIFO is available.
+            - When available, confirm that the TPM submodule sends START followed by the register value.
+            - Compare this value with the expected value.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tpm_write
+      desc: '''
+            - Set TPM_CFG.TPM_MODE to 0.
+            - Set TPM_CFG.EN.
+            - Randomise other fields in TPM_CFG.
+            - Send TPM write command with a randomised address.
+            - Check TPM_CMD_ADDR.
+            - Based on FIFO status, check SPI bus to confirm WAIT or START sent.
+            - Check that the TPM submodule accepts write data without the WAIT state if the write FIFO is empty.
+            - Otherwise, check WAIT until the write FIFO becomes available (empty).'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: tpm_crb_read
+      desc: '''
+            Set TPM_CFG.TPM_MODE to 1.
+            Set TPM_CFG.EN.
+            Randomise other fields in TPM_CFG.
+            Send TPM read command over the SPI bus with a randomised address.
+            Check TPM_STATUS.cmdaddr_notempty, INTR_STATE.tpm_header_notempty.
+            Confirm that invalid locality check still runs based on the invalid_locality configuration.
+            Confirm that the TPM submodule sends START followed by the register value.
+            Compare this value with the expected value.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: partial_txrx
+      desc: '''
+            - Verify that when timer expires while bytes are still in the
+              RXFIFO, the logic writes partial words to SRAM.
+            - Confirm read-modify-write operation is triggered to perform the
+              partial update.
+            - Perform multiple partial TX/RX.'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: sck_config
+      desc: '''
+            - Verify that the SPI clock and phase (CFG.CPOL, CFG.CPHA) behave
+              as expected by performing a generic rw transaction with
+              combinations of polarity and phase.
+            - Verify that in the configuration (CFG.CPOL=1, CFG.CPHA=1), the
+              logic isn't able to pop the entry from the TX async FIFO after
+              the last bit in the last byte of a transaction.
+            - Confirm that if CSB is de-asserted in bit 1 to 6 position, the
+              FIFO entry isn't popped and that TX logic will re-send the byte
+              in next transaction.
+            - Confirm that if CSB is de-asserted in the 7th or 8th bit
+              position, the data is dropped and will re-commence with the next
+              byte in the next transaction.'''
```

Comment:
Maybe I am wrong, but if CSb deasserted at the 7th beat, then FIFO entry won't pop. Only after 7th negedge, the fifo entry will be consumed.